react/fields/FieldMessage/FieldMessage.js

```diff
@@ -0,0 +1,119 @@
+import styles from './FieldMessage.less';
+
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+
+import ErrorIcon from '../../icons/ErrorIcon/ErrorIcon';
+import TickCircleIcon from '../../icons/TickCircleIcon/TickCircleIcon';
+
+import Text from '../../Text/Text';
+
+function combineClassNames(props = {}, ...classNames) {
+  const { className, ...restProps } = props;
+
+  return {
+    className: classnames.apply(null, [...classNames, className]), // eslint-disable-line no-useless-call
+    ...restProps
+  };
+}
+
+export default class FieldMessage extends Component {
+
+  static displayName = 'FieldMessage';
+
+  static propTypes = {
+    invalid: PropTypes.bool,
+    help: React.PropTypes.string,
+    helpProps: PropTypes.object,
+    valid: PropTypes.bool,
+    message: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.oneOf([false])
+    ]),
+    messageProps: PropTypes.object
+  };
+
+  static defaultProps = {
+    message: '',
+    messageProps: {
+      critical: false,
+      positive: false,
+      secondary: false
+    }
+  };
+
+  constructor() {
+    super();
+
+    this.renderMessage = this.renderMessage.bind(this);
+    this.renderMessageIcon = this.renderMessageIcon.bind(this);
+  }
+
+  renderMessage() {
+    const { message, valid } = this.props;
+
+    if (message) {
+      const { critical, positive, secondary, ...restMessageProps } = this.props.messageProps;
+      const allMessageProps = combineClassNames(restMessageProps, styles.message);
+
+      return (
+        <Text
+          {...allMessageProps}
+          critical={(valid === false && !secondary) || critical}
+          positive={(valid === true && !secondary) || positive}
+          secondary={typeof valid === 'undefined' || secondary}>
+          {this.renderMessageIcon()}
+          {message}
+        </Text>
+      );
+    }
+
+    return null;
+  }
+
+  renderMessageIcon() {
+    const { valid } = this.props;
```

Comment:
You have a mix of `valid` and `invalid` in this component, but only one present in the `propTypes`